File: app/src/main/java/com/evans/technologies/usuario/Activities/cupon/PagosActivity.java
package com.evans.technologies.usuario.Activities.cupon;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.evans.technologies.usuario.R;
import com.evans.technologies.usuario.Utils.Adapters.adapter_rv_cupon;
import com.evans.technologies.usuario.model.DataCupon;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PagosActivity extends AppCompatActivity implements View.OnClickListener{
    @BindView(R.id.ap_rv_metodos_pay)
    RecyclerView metodos_pay;
    @BindView(R.id.ap_txt_add_pagos)
    TextView add_pay;
    @BindView(R.id.ap_txt_add_promo)
    TextView add_promo;
    @BindView(R.id.ap_ll_cupones)
    LinearLayout cupones;
    @BindView(R.id.ap_imagebutton_back)
    ImageButton back;
    private RecyclerView.Adapter adapterRview;
    ArrayList<DataCupon> tipepays= new ArrayList<>();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pagos);
        ButterKnife.bind(this);
        DataCupon efectivo= new DataCupon("Cesar Willy","efectivo","Luis rivarola","Puno","Puno");
        tipepays.add(efectivo);
        metodos_pay.setLayoutManager( new LinearLayoutManager(PagosActivity.this) );

        adapterRview = new adapter_rv_cupon(this, R.layout.dialog_rv_metodos_pago, "metodopay", tipepays, new adapter_rv_cupon.click() {
            @Override
            public void itemClick(DataCupon data, int adapterPosition) {

            }
        });
        add_pay.setOnClickListener(this);
        add_promo.setOnClickListener(this);
        cupones.setOnClickListener(this);
        metodos_pay.setAdapter(adapterRview);
        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.ap_txt_add_pagos:
                break;
            case R.id.ap_txt_add_promo:
                startActivity(new Intent(PagosActivity.this,AddCodePromocional.class));
                break;
            case R.id.ap_ll_cupones:
                startActivity(new Intent(PagosActivity.this,Cupones.class));
                break;
        }
    }
}

File: app/src/main/java/com/evans/technologies/usuario/Activities/cupon/PagosActivity.kt
package com.evans.technologies.usuario.Activities.cupon

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.evans.technologies.usuario.R
import com.evans.technologies.usuario.Utils.Adapters.adapter_rv_cupon
import com.evans.technologies.usuario.Utils.Adapters.adapter_rv_cupon.click
import com.evans.technologies.usuario.model.DataCupon
import kotlinx.android.synthetic.main.activity_pagos.*
import java.util.*

class PagosActivity : Fragment() {


    private var adapterRview: RecyclerView.Adapter<*>? = null
    var tipepays = ArrayList<DataCupon>()
    val efectivo =
        DataCupon("Cesar Willy", "efectivo", "Luis rivarola", "Puno", "Puno")
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.activity_pagos, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (tipepays!=null)
            tipepays.clear()
        tipepays.add(efectivo)

        ap_rv_metodos_pay!!.layoutManager = LinearLayoutManager(requireContext())
        val adapterclass = adapter_rv_cupon(
            requireContext(),
            R.layout.dialog_rv_metodos_pago,
            click { data, adapterPosition ->

            })
//        ap_txt_add_pagos!!.setOnClickListener(this)
        adapterRview=adapterclass
        ap_ll_cupones!!.setOnClickListener{
            findNavController().navigate(R.id.action_nav_wallet_to_cupones)
        }
        adapterclass.getData(tipepays, "metodopay")

        ap_rv_metodos_pay!!.adapter = adapterRview
    }



}